backend/src/schema/resolvers/posts.js

```diff
@@ -211,10 +225,75 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPostWithNestedAttributes
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH (:User)-[previousRelations:PINNED]->(post:Post)
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-18T13:12:36Z" title="Friday, October 18th 2019, 3:12:36 pm +02:00">Oct 18, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -211,10 +225,68 @@ export default {
       })
       return emoted
     },
+    pinPost: async (_parent, params, context, _resolveInfo) => {
+      let pinnedPost
+      const { driver, user } = context
+      const session = driver.session()
+      const { id: userId } = user
+      let writeTxResultPromise = session.writeTransaction(async transaction => {
+        const deletePreviousRelationsResponse = await transaction.run(
+          `
+          MATCH ()-[previousRelations:PINNED]->(post:Post)
```


Why not `User`?
```suggestion
          MATCH (:User)-[previousRelations:PINNED]->(post:Post)
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-18T16:18:38Z" title="Friday, October 18th 2019, 6:18:38 pm +02:00">Oct 18, 2019</time>_

---


Cause it's not important what or who pinned it

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-10-19T08:40:02Z" title="Saturday, October 19th 2019, 10:40:02 am +02:00">Oct 19, 2019</time>_

---


???
Why then in file `backend/src/schema/types/type/Post.gql`, line 19:

```
pinnedAt: String @cypher(
      statement: "MATCH (this)<-[pinned:PINNED]-(:User) WHERE NOT this.deleted = true AND NOT this.disabled = true RETURN pinned.createdAt"
    )
```

In my eyes it is a kind of code quality if we are as good as specific we can be. That’s why we are writing test to nail the things more down, as an example.

For sure at the moment only users can pin posts. But that only users can do something is true  for a lot of things. Only users can emote to posts, but we query for the label `User` as well like in file "backend/src/schema/resolvers/posts.js“, line 315:
```
        emotionsCount: '<-[related:EMOTED]-(:User)',
```

I thing we should just decide if we want to be as specific as possible or as free as possible …
In my eyes more specific tracks down more possible errors …

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-10-19T09:40:26Z" title="Saturday, October 19th 2019, 11:40:26 am +02:00">Oct 19, 2019</time>_

---


Sure, I updated it already... 
I don't think it's important, but if you do, I'm happy to update it, as I showed with my actions.